src/osd/PG.cc

```diff
@@ -3031,6 +3031,14 @@ void PG::scrub_compare_maps()
   for (const auto map : maps) {
     for (const auto i : map.second->objects) {
       master_set.insert(i.first);
+
+      if (i.second.size > cct->_conf->osd_max_object_size) {
+	stringstream ss;
+	ss << info.pgid << " " << i.first << " size " << i.second.size
+	   << " > osd_max_object_size " << cct->_conf->osd_max_object_size;
+	++scrubber.shallow_errors;
+	osd->clog->error(ss);
+      }
```

Comment:
@liewegas Doing the check here will check every replica of every object.  This will count every copy of an object that is too large instead of counting distinct objects.  Also, it creates an error count that doesn't correspond with list-inconsistent-obj output.  I can create a commit that adds a new object error flag.  It would be typical that all replicas have the same size, so it would be nice to add an object level error flag and count this "too large" error once per object if any replica is too large.